presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/converter/GlueToPrestoConverter.java

```diff
@@ -75,81 +76,166 @@ public static Table convertTable(com.amazonaws.services.glue.model.Table glueTab
-                        .collect(toImmutableList()))
+                .setDataColumns(convertColumns(sd.getColumns()))
                 .setParameters(tableParameters)
                 .setViewOriginalText(Optional.ofNullable(glueTable.getViewOriginalText()))
                 .setViewExpandedText(Optional.ofNullable(glueTable.getViewExpandedText()));
 
         if (glueTable.getPartitionKeys() != null) {
-            tableBuilder.setPartitionColumns(glueTable.getPartitionKeys().stream()
-                    .map(GlueToPrestoConverter::convertColumn)
-                    .collect(toImmutableList()));
+            tableBuilder.setPartitionColumns(convertColumns(glueTable.getPartitionKeys()));
         }
         else {
-            tableBuilder.setPartitionColumns(new ArrayList<>());
+            tableBuilder.setPartitionColumns(ImmutableList.of());
         }
-
-        setStorageBuilder(sd, tableBuilder.getStorageBuilder(), tableParameters);
+        // No benefit to memoizing here, just reusing the implementation
+        new StorageConverter().setStorageBuilder(sd, tableBuilder.getStorageBuilder(), tableParameters);
         return tableBuilder.build();
     }
 
-    private static void setStorageBuilder(StorageDescriptor sd, Storage.Builder storageBuilder, Map<String, String> tableParameters)
+    private static Column convertColumn(com.amazonaws.services.glue.model.Column glueColumn)
     {
-        requireNonNull(sd.getSerdeInfo(), "StorageDescriptor SerDeInfo is null");
-        SerDeInfo serdeInfo = sd.getSerdeInfo();
+        return new Column(glueColumn.getName(), HiveType.valueOf(glueColumn.getType().toLowerCase(Locale.ENGLISH)), Optional.ofNullable(glueColumn.getComment()));
+    }
 
-        Optional<HiveBucketProperty> bucketProperty = Optional.empty();
-        if (sd.getNumberOfBuckets() > 0) {
-            if (isNullOrEmpty(sd.getBucketColumns())) {
-                throw new PrestoException(HIVE_INVALID_METADATA, "Table/partition metadata has 'numBuckets' set, but 'bucketCols' is not set");
-            }
-            List<SortingColumn> sortedBy = ImmutableList.of();
-            if (!isNullOrEmpty(sd.getSortColumns())) {
-                sortedBy = sd.getSortColumns().stream()
-                        .map(column -> new SortingColumn(
-                                column.getColumn(),
-                                Order.fromMetastoreApiOrder(column.getSortOrder(), "unknown")))
-                        .collect(toImmutableList());
-            }
-            BucketingVersion bucketingVersion = HiveBucketing.getBucketingVersion(tableParameters);
-            bucketProperty = Optional.of(new HiveBucketProperty(sd.getBucketColumns(), bucketingVersion, sd.getNumberOfBuckets(), sortedBy));
+    private static List<Column> convertColumns(List<com.amazonaws.services.glue.model.Column> glueColumns)
+    {
+        ImmutableList.Builder<Column> builder = ImmutableList.builderWithExpectedSize(glueColumns.size());
```

Comment:
You could use a stream here

Reply:
You could, but for simple transformations like this that are relatively common and when the output list size is guaranteed to be known, I've found defaulting to a sized list to be preferable. See my reply about the `buildPartitionNames` case.

Obviously, 1M columns is much less likely than 1M partitions, but `convertColumns` is also performed within each partition conversion so the effect can be much more significant when the number of columns is non-trivial.

Reply:
So maybe we should improve `io.prestosql.util.MoreLists#mappedCopy` and make it more commonly used?

Reply:
@findepi that's a good suggestion. We could do that here and have even more concise code than streams
```java
return mappedCopy(glueColumns, this::convertColumn);

Reply:
To be clear --  while i am concerned about conciseness, i am even more concerned about "durability" of the change.
builder-based code can easily be later replaced with stream-based in a wholesale fashion. "Improve readability", "Use JDK APIs for mapping lists", etc can be compelling reason to merge such a change. Given we cannot test improvement proposed here, we need to find a non-test way to make it stay with us for longer.
Thus, using `mappedCopy` alone is not enough. We need to also adorn that method with some explanation why&where it is better than stream/map/list.

Reply:
Added a comment in to the `mappedCopy` method explaining the presized builder.